From e8196d55ecac7f2bf3aa5cb0aefef0122e63db21 Mon Sep 17 00:00:00 2001
From: Michael Smith <michael.smith@puppet.com>
Date: Thu, 13 Oct 2016 14:45:13 -0700
Subject: [PATCH] (LTH-115) Fix Boost.Log sink initialization with Boost 1.62

In Boost 1.62, the way sink argument forwarding appears to have changed
in such a way that it failed to identify the implicit creation of a
color_writer. Switch to explicitly creating the color_writer sink.
---
 logging/src/logging.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/logging/src/logging.cc b/logging/src/logging.cc
index 9c071dc..d7419d5 100644
--- logging/src/logging.cc
+++ logging/src/logging.cc
@@ -82,7 +82,7 @@ namespace leatherman { namespace logging {
         core->remove_all_sinks();
 
         using sink_t = sinks::synchronous_sink<color_writer>;
-        boost::shared_ptr<sink_t> sink(new sink_t(&dst));
+        boost::shared_ptr<sink_t> sink = boost::make_shared<sink_t>(boost::make_shared<color_writer>(&dst));
         core->add_sink(sink);
 
 
